Cogburn/session to case

coreyogburn opened the pull request:
Allow attaching shared sessions to existing or new cases. If attached to a case, the session cannot be unshared.

coreyogburn committed:
Attach Session To Case

Add an escalate triangle to the assistant page, only show it when a session has been loaded. Clicking it offers the same mruCases list we use in Alerts including creating a new case.

Case has been updated to show Assistant Sessions slightly differently from other attachments such as hiding hashes.

To do: Still need to modify UpdateSession endpoint to not allow the removal of the "shared" tag if the session is attached to a case. Tests.

coreyogburn committed:
Checks, Errors, Cleanup, Tests

New case store function GetCaseIdsWithArtifact hunts for cases that have the artType and value provided, especially useful for checking if a case has a specific assistant session attached.

When unsharing a session, the backend now checks if the session has been attached to any cases. If so, a custom error is returned.

Fixed a missing i18n entry.

Added tests.

coreyogburn committed:
Hide Protected Toggle

coreyogburn committed:
Rebase fix

After rebasing and resolving conflicts, an extra closing bracket needed removing.

coreyogburn committed:
Fix Test

jertel commented on html/js/routes/assistant.js:
```diff
@@ -1779,6 +1781,8 @@ routes.push({ path: '/assistant/:sessionId?', name: 'assistant', component: {
       if (localStorage[prefix + '.showChatHistory']) this.showChatHistory = localStorage[prefix + '.showChatHistory'] == 'true';
       if (localStorage[prefix + '.currentModel']) this.currentModel = localStorage[prefix + '.currentModel'];
+
+			if (localStorage['settings.case.mruCases']) this.mruCases = JSON.parse(localStorage['settings.case.mruCases']);
```
Is there a tab/indent mismatch here?

coreyogburn replied:
Yes. Fixed, and double checked IDE settings to use 2 spaces.

jertel commented on html/pages/assistant.html:
```diff
@@ -110,6 +110,34 @@ <h5 class="ml-5 mt-2" data-aid="assistant_current_model">{{ modelsMap.get(curren
 						fa-file-export
 					</v-icon>
+					<template v-if="currentChatId">
+						<v-icon class="mx-2" v-if="currentChatId" @click="toggleSharedSession(currentChatId)" data-aid="assistant_banner_share_toggle" :disabled="$root.user.id !== chatHistoryById?.[currentChatId]?.userId" :title="i18n.assistantLinkTitle">
```
Consider using `$root.isMyUser(...?.userId)` here

coreyogburn replied:
Thanks, didn't know about that one. I need to browse app.js and refresh myself.

coreyogburn committed:
Respond to PR Feedback

Use $root.isMyUser(...) for a more readable condition.

Fix tab/space indention mismatch.

jertel commented on html/pages/assistant.html:
```diff
@@ -111,7 +111,7 @@ <h5 class="ml-5 mt-2" data-aid="assistant_current_model">{{ modelsMap.get(curren
 					</v-icon>
 					<template v-if="currentChatId">
-						<v-icon class="mx-2" v-if="currentChatId" @click="toggleSharedSession(currentChatId)" data-aid="assistant_banner_share_toggle" :disabled="$root.user.id !== chatHistoryById?.[currentChatId]?.userId" :title="i18n.assistantLinkTitle">
+						<v-icon class="mx-2" v-if="currentChatId" @click="toggleSharedSession(currentChatId)" data-aid="assistant_banner_share_toggle" :disabled="$root.isMyUser(chatHistoryById?.[currentChatId]?.userId)()" :title="i18n.assistantLinkTitle">
```
Is this extra `()` correct?

coreyogburn replied:
Nope, removed.

coreyogburn committed:
Correct Typo
